update SDK to 0.33.0

bcressey opened the pull request:
**Issue number:**
N/A

**Description of changes:**
Switch to the new SDK, which includes:
* Rust 1.70.0
* Go 1.20.5
* `xfsprogs`
* re-homed RPM macros

The macro move requires some minor fixes. The most visible change will be that RPMs now have the architecture set correctly to the target's architecture, rather than including that in the package name. RPMs aren't directly user-facing except in the case of `application-inventory.json`, which is now simpler to generate.

I've also cleaned up packages that were (incorrectly) tagged as "noarch". This changes `application-inventory.json` for the affected packages, which move from `"Architecture": "noarch"` to the target's architecture.

**Testing done:**
Built some variants:
* aws-k8s-1.24 on x86_64 host for x86_64 target
* aws-k8s-1.24 on aarch64 host for aarch64 target
* aws-k8s-1.27 on x86_64 host for aarch64 target
* aws-k8s-1.27 on aarch64 host for x86_64 target

I checked `application-inventory.json` on the running instances and confirmed that the prefix was correctly matched and removed, and that the "Architecture" field was set to the target's architecture for all packages.


**Terms of contribution:**

By submitting this pull request, I agree that this contribution is dual-licensed under the terms of both the Apache License, version 2.0, and the MIT license.

Commit:
build: drop "BuildArch: noarch" from all packages

This was used as a quick way to disable debuginfo generation, but it
creates ambiguity when using the RPM's architecture to indicate what
target the package was built for.

These packages capture the architecture in the sysroot-based paths
used for installed files, so they are not actually arch-agnostic in
the sense that "noarch" implies.

Signed-off-by: Ben Cressey <bcressey@amazon.com>

Commit:
build: set target CPU to target arch for RPMs

This causes the RPM package architecture to match the target's rather
than the host's. By default, `dnf` and `rpm` will ignore non-native
RPMs, but both can be forced to use them.

When the package architecture is set correctly, it's not necessary to
include the architecture in the package name, and the macros in the
SDK no longer do so. Fix up various places in the image build process
that assumed the architecture would be present.

Signed-off-by: Ben Cressey <bcressey@amazon.com>

Commit:
build: update SDK to 0.33.0

Signed-off-by: Ben Cressey <bcressey@amazon.com>

Commit:
build: use macros from SDK

Signed-off-by: Ben Cressey <bcressey@amazon.com>

Commit:
grub: switch to new arch macros

Signed-off-by: Ben Cressey <bcressey@amazon.com>

Commit:
sources: fix clippy lints for rust 1.70.0

Signed-off-by: Ben Cressey <bcressey@amazon.com>

stmcginnis commented:
Please revert ee7205d92c7f8aaeb4ea8d66105f158c9731fc6c in this PR.

We had to pin the node's rust version temporarily, otherwise the cargo index format was getting a mismatch between the 1.69 format and the newer 1.70 sparse format. Once we move to 1.70 in the SDK, we can remove that pin so the host gets the `latest` version, which will be 1.70 or later that supports the right format.

Commit:
build: unpin Rust version

This reverts commit ee7205d92c7f8aaeb4ea8d66105f158c9731fc6c.

Pinning the Rust version is no longer required, now that Rust 1.70 is
in the SDK and supports the new sparse index format by default.

stmcginnis reviewed (approved):
Looks good to me.
